var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -80,15 +80,16 @@ class Cuda(Package):
     depends_on('libxml2', when='@10.1.243:')
 
     def setup_build_environment(self, env):
-        env.set('CUDAHOSTCXX', self.compiler.cxx)
         if self.spec.satisfies('@10.1.243:'):
             libxml2_home  = self.spec['libxml2'].prefix
             env.set('LIBXML2HOME', libxml2_home)
             env.append_path('LD_LIBRARY_PATH', libxml2_home.lib)
 
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('CUDAHOSTCXX', self.compiler.cxx)
```

Comment:
should probably be the compiler of the dependent spec?

Reply:
yes

Reply:
```suggestion
        env.set('CUDAHOSTCXX', dependent_spec.package.compiler.cxx)
```